llvm/unittests/ExecutionEngine/Orc/CoreAPIsTest.cpp

```diff
@@ -1559,16 +1559,13 @@ TEST_F(CoreAPIsStandardTest, TestLookupWithThreadedMaterialization) {
 #if LLVM_ENABLE_THREADS
 
   std::mutex WorkThreadsMutex;
-  std::vector<std::thread> WorkThreads;
+  SmallVector<std::thread,0> WorkThreads;
   DispatchOverride = [&](std::unique_ptr<Task> T) {
-    std::promise<void> WaitP;
-    std::lock_guard<std::mutex> Lock(WorkThreadsMutex);
+    std::unique_lock Lock(WorkThreadsMutex);
```

Comment:
I think this could remain a `lock_guard`, since manual locking/unlocking isn't needed here.

Reply:
That is true, but it just seemed more simple to use the same type everywhere: I don't actually know why `std::lock_guard` exists when `std::unique_lock` appears to be simply better than it in every way?

Reply:
FWIW, it looks like LLVM now may consider `lock_guard` deprecated in favor of drive-by replacements of it with `unique_lock` everywhere? #126434 #146240

Reply:
Yeah, I am made slightly uncomfortable by using two completely different RAII locks on the same mutex, though I realize that it is well-defined, so I dropped this part of the chagne. The other changes are all related to the bug fix, so it isn't particularly easy to justify separating them (SmallVector is needed because the iteration structure needed to change and the relevant feature is part of SmallVector not std::vector)